pkg/util/parameters.go

```diff
@@ -0,0 +1,180 @@
+package util
+
+import (
+	"context"
+	"fmt"
+	"sync"
+
+	ackerr "github.com/aws-controllers-k8s/runtime/pkg/errors"
+)
+
+var (
+	errUnknownParameter      = fmt.Errorf("unknown parameter")
+	errUnmodifiableParameter = fmt.Errorf("parameter is not modifiable")
+)
+
+func NewErrUnknownParameter(name string) error {
+	// This is a terminal error because unless the user removes this parameter
+	// from their list of parameter overrides, we will not be able to get the
+	// resource into a synced state.
+	return ackerr.NewTerminalError(
+		fmt.Errorf("%w: %s", errUnknownParameter, name),
+	)
+}
+
+func NewErrUnmodifiableParameter(name string) error {
```

Comment:
Public functions/objects needs to have a comment - https://github.com/golang/go/wiki/CodeReviewComments#doc-comments

Reply:
Done